src/tools/rbd/action/MirrorPool.cc

```diff
@@ -259,16 +259,48 @@ void get_enable_arguments(po::options_description *positional,
 }
 
 int execute_enable_disable(const std::string &pool_name,
-                           rbd_mirror_mode_t mirror_mode) {
+                           const std::string &mode) {
   librados::Rados rados;
   librados::IoCtx io_ctx;
+  rbd_mirror_mode_t next_mirror_mode;
+  rbd_mirror_mode_t current_mirror_mode;
+
+  if (mode == "disable") {
+    next_mirror_mode = RBD_MIRROR_MODE_DISABLED;
+  } else if (mode == "image") {
+    next_mirror_mode = RBD_MIRROR_MODE_IMAGE;
+  } else if (mode == "pool") {
+    next_mirror_mode = RBD_MIRROR_MODE_POOL;
+  } else {
+    std::cerr << "rbd: must specify 'image' or 'pool' mode." << std::endl;
+    return -EINVAL;
+  }
+
   int r = utils::init(pool_name, &rados, &io_ctx);
   if (r < 0) {
     return r;
   }
 
   librbd::RBD rbd;
-  r = rbd.mirror_mode_set(io_ctx, mirror_mode);
+  r = rbd.mirror_mode_get(io_ctx, &current_mirror_mode);
+  if (r < 0) {
+    std::cerr << "rbd: failed to retrieve mirror mode: " << cpp_strerror(r)
+              << std::endl;
+    return r;
+  }
+
+  if (current_mirror_mode == next_mirror_mode) {
+    std::cout << "mirroring mode on pool is already " << mode << std::endl;
```

Comment:
This is still lightly off grammar-wise. If disabling, the text should be something like "mirroring is already disabled", otherwise it can be something like "mirroring is already configured for <mode> mode".